src/routes/calculator/components/Calculator.js

```diff
@@ -0,0 +1,47 @@
+ import React from 'react';
+ import PropTypes from 'prop-types';
+ import './calculator.scss';
+
+
+class Calculator  extends React.Component {
+    constructor(props){
+       super(props)
+     }
+
+    render() {
+        const { calcState, addElem, clear, equal } = this.props;
+        return (
+             <div className="App">
+               <div className="value-container">
+                 <input type="text" value={calcState.calValue} />
+               </div>
+               <div className="buttons-container">
+                    {calcState.calBtns.map((item, key) => {
+                        if(item == "C"){
+                            return(
+                               <button onClick={clear.bind(this) } key={key}>{item}</button>
```

Comment:
You don't have to bind this(everywhere in this render) I think, try removing it and see